src/components/map/JourneyPosition.js

```diff
@@ -31,58 +36,77 @@ class JourneyPosition extends Component {
         []
       );
 
-      const pos = getCoarsePositionForTime(journeyPositions, timeDate, journeyId);
+      const pos = await getCoarsePositionForTime(
```

Comment:
This now uses the async animationFrame helper, so lots of change had to be done to accommodate an async value-